### ISSUE TITLE
Graphic bugs after changing deqeueReusableAnnotationImage

### ISSUE THREAD

**Issue by moax** [mapbox/mapbox-gl-native#9323]
**Platform: IOS**
**Mapbox SDK version: 3.5.4**

After changing the image of a deqeueReusableImage the image is drawn buggy (see image attached).
Following code is used:

```
 func mapView(_ mapView: MGLMapView, imageFor annotation: MGLAnnotation) -> MGLAnnotationImage? {

  var annotationImage = MapView.dequeueReusableAnnotationImage(withIdentifier: identifierForAngle)
            if annotationImage == nil {
                var image = UIImage(named: "airportHard"+identifierForAngle)!
                annotationImage = MGLAnnotationImage(image: image, reuseIdentifier: identifierForAngle)
            }    return annotationImage  }

func mapView(_ mapView: MGLMapView, regionDidChangeAnimated animated: Bool) {
       
if (MapView.zoomLevel < 8.0 && airportIconsDetailed == true) {
           var image = UIImage(named: "airportFar")!
           self.MapView.dequeueReusableAnnotationImage(withIdentifier: "90")?.image = image
}}
```

I tried a different approach width removing the Annotation and adding it again, but the function mapView(_ mapView: MGLMapView, imageFor annotation: MGLAnnotation) is not called then and the image is the standard marker. 

### Expected behavior
Changing the Image as it should be.

### Actual behavior
before changing image
![bildschirmfoto 2017-06-21 um 14 35 24](https://user-images.githubusercontent.com/2041717/27384374-eed625ba-568e-11e7-9561-a08e761ca33a.png)

after changing image
![bildschirmfoto 2017-06-21 um 14 35 35](https://user-images.githubusercontent.com/2041717/27384389-f7383dce-568e-11e7-87c0-b1aff493051e.png)

**Comment by fabian-guerra** [comment]
I tested this with the release branch and I'm able to reproduce the issue. Using `master` I'm not, so looks like it was fixed there. Do you know which commit it may be related so I can cherry-pick to release? @jfirebaugh @kkaefer

**Comment by jfirebaugh** [comment]
#9213, but it's a complex change with many prerequisites; it's not feasible to cherry-pick it to the 5.1.0 release branch. We'll need to wait until the next release from `master`.

**Comment by fabian-guerra** [comment]
Thanks @jfirebaugh.

**Comment by moax** [comment]
When do you expect the next release from master? I am unable to publish my project with this kind of bug..

**Comment by fabian-guerra** [comment]
Hi @moax, we don't have a release date yet. You may try [`MGLAnnotationView`](https://www.mapbox.com/ios-sdk/examples/annotation-views/) or [`MGLSymbolsStyleLayer`](https://www.mapbox.com/ios-sdk/examples/runtime-multiple-annotations/) as a workaround instead.

**Comment by jfirebaugh** [comment]
Closing as fixed; this is fixed for 3.7.0 and we're not planning to backport the fix.
